jobs/sql/extractions/all_encounters_staging.sql

```diff
@@ -14,49 +14,54 @@ user_entered        varchar(255),
 encounter_location  varchar(255),
 encounter_type_name varchar(50),
 entered_datetime    datetime,
+combined_date_changed	datetime,
 emr_id              varchar(15),
 next_appt_date      date,
 voided				bit
 );
 
-select last_loaded_time into @lastloadtime from petl_load_times where loaded_domain = 'encounter';
+set @last_loaded_max_time = null;
+select max(loaded_max_datetime) into @last_loaded_max_time from petl_load_times where loaded_domain = 'encounter' and status = 'complete';
 
-Insert into temp_all_encounters (
-encounter_id, encounter_datetime, encounter_type_name, encounter_location, patient_id, visit_id, user_entered, entered_datetime, voided)
-select encounter_id, encounter_datetime, encounter_type_name_from_id(encounter_type), location_name(location_id), patient_id, visit_id, person_name_of_user(creator), date_created, voided 
+insert into temp_all_encounters (
+encounter_id, encounter_datetime, encounter_type_name, encounter_location, patient_id, visit_id, user_entered, entered_datetime, voided, combined_date_changed)
+select encounter_id, encounter_datetime, encounter_type_name_from_id(encounter_type), location_name(location_id), patient_id, visit_id, person_name_of_user(creator), date_created, voided,
+CASE 
+	when ifnull(date_created,'1000-01-01') >= ifnull(date_changed,'1000-01-01') and ifnull(date_created,'1000-01-01') >= ifnull(date_voided,'1000-01-01') then date_created
+	when ifnull(date_changed,'1000-01-01') >= ifnull(date_created,'1000-01-01') and ifnull(date_changed,'1000-01-01') >= ifnull(date_voided,'1000-01-01') then date_changed
+	else date_voided
+END 
```

Comment:
Might be nicer to use "GREATEST" function here, like `greatest(ifnull(date_created,'1000-01-01'), ifnull(date_changed,'1000-01-01'), ifnull(date_voided,'1000-01-01'))`